pkg/clean/clean.go

```diff
@@ -27,19 +27,25 @@ var log = logger.Log
 // including security policy, static route, subnet, subnet port, subnet set, vpc, ip pool, nsx service account
 // it is usually used when nsx-operator is uninstalled and remove all the resources created by nsx-operator
 // return error if any, return nil if no error
-func Clean(cf *config.NSXOperatorConfig) (Status, error) {
+// the error type include followings:
+//
+//	 	ValidationFailed 			indicate that the config is incorrect and failed to pass validation
+//		GetNSXClientFailed  		indicate that could not retrieve nsx client to perform cleanup operation
+//		InitCleanupServiceFailed 	indicate that error happened when trying to initialize cleanup service
+//		CleanupResourceFailed    	indicate that the cleanup operation failed at some services, the detailed will in the service logs
+func Clean(cf *config.NSXOperatorConfig) error {
 	log.Info("starting NSX cleanup")
 	if err := cf.ValidateConfigFromCmd(); err != nil {
-		return ValidationFailed, err
+		return ValidationFailed
 	}
 	nsxClient := nsx.GetClient(cf)
 	if nsxClient == nil {
-		return GetNSXClientFailed, fmt.Errorf("failed to get nsx client")
+		return GetNSXClientFailed
 	}
 	if cleanupService, err := InitializeCleanupService(cf); err != nil {
-		return InitCleanupServiceFailed, fmt.Errorf("failed to initialize cleanup service: %w", err)
+		return InitCleanupServiceFailed
```

Comment:
I think it would be valuable to wrap the original error. You know better about this than I with respect to the internals of these packages,  but Go supports error wrapping so we can validate the specific type returned, but also keep context of deeper stack errors.

See https://pkg.go.dev/errors. In particular, you can make easy use of `errors.Join` - see https://pkg.go.dev/errors#example-Join.

ex:

```go
return errors.Join(InitCleanupServiceFailed, err)
```

Consumers can then use this for logging or deeper inspection if they desire.